data/trie/patriciaMerkleTrie.go

```diff
@@ -40,6 +50,7 @@ func NewTrie(
 	hsh hashing.Hasher,
 	evictionDb storage.Persister,
 	evictionWaitListSize int,
+	snapshotDbCfg config.DBConfig,
```

Comment:
Config parsing should happen on main / struct.go - this constructor should get the result of the config - like the actual file in this case.

Reply:
I can't pass a database to the constructor of the trie, because each snapshot is stored in a separate database. Each time snapshot is called, a new database needs to be created, so I need the config parameters to be able to create a new database.